stl/inc/flat_map

```diff
@@ -744,6 +744,24 @@ public:
         return {lower_bound(_X), upper_bound(_X)};
     }
 
+    friend bool operator==(const flat_map& _X, const flat_map& _Y) {
+        return _RANGES equal(_X._Data.keys, _Y._Data.keys) && _RANGES equal(_X._Data.values, _Y._Data.values);
+    }
+
+    friend _STD weak_ordering operator<=>(const flat_map& _X, const flat_map& _Y) {

```

Comment:
This is not correct. See https://eel.is/c++draft/container.opt.reqmts

Reply:
I didn't know the existence of `_Synth_three_way`. Looks like I can just use that.

Reply:
Fixed.